plugins/readtimer.koplugin/main.lua

```diff
@@ -99,72 +110,62 @@ function ReadTimer:remainingTime(round)
     end
 end
 
-function ReadTimer:addAdditionalHeaderFooterContent(no_event)
-    if not self.ui.crelistener then return end
-
-    local timer_symbol = "\u{23F2}"  -- ⏲ timer symbol
-    local timer_letter = "T"
-
-    self.ui.crelistener:removeAdditionalHeaderContent(timer_symbol)
-    self.ui.view.footer:removeAdditionalFooterContent(timer_symbol)
-    self.ui.view.footer:removeAdditionalFooterContent(timer_letter)
-
-    if not self:scheduled() or (not self.show_timer_value_in_header and not self.show_timer_value_in_footer) then
-        if not no_event then
-            UIManager:broadcastEvent(Event:new("UpdateHeader"))
-            UIManager:broadcastEvent(Event:new("UpdateFooter", true))
-        end
-        return
-    end
-
-    local hours, minutes, dummy = self:remainingTime(1)
-    local timer_info = string.format("%02d:%02d", hours, minutes)
-    UIManager:scheduleIn(60, ReadTimer.addAdditionalHeaderFooterContent, self)
+local timer_symbol = "\u{23F2}"  -- ⏲ timer symbol
+local timer_letter = "T"
 
-    if self.show_timer_value_in_header then
-        self.ui.crelistener:addAdditionalHeaderContent(timer_symbol .. timer_info .. " ")
-    end
-    if not no_event then
-        UIManager:broadcastEvent(Event:new("UpdateHeader"))
+local function additional_header_content_func(this)
+    if this:scheduled() then
+        local hours, minutes, dummy = this:remainingTime(1)
+        local timer_info = string.format("%02d:%02d", hours, minutes)
+        return timer_symbol .. timer_info .. " "
+    else
+        return ""
     end
+end
+local function additional_footer_content_func(this)
+    if this:scheduled() then
+        local item_prefix = this.ui.view.footer.settings.item_prefix
+        local hours, minutes, dummy = this:remainingTime(1)
+        local timer_info = string.format("%02d:%02d", hours, minutes)
 
-    if self.show_timer_value_in_footer then
-        local item_prefix = self.ui.view.footer.settings.item_prefix
         if item_prefix == "icons" then
-            timer_info = timer_symbol .. " " .. timer_info
+            return timer_symbol .. " " .. timer_info
         elseif item_prefix == "compact_items" then
-            timer_info = timer_symbol .. timer_info
+            return timer_symbol .. timer_info
         else
-            timer_info = timer_letter .. ": " .. timer_info
+            return timer_letter .. ": " .. timer_info
         end
-        self.ui.view.footer:addAdditionalFooterContent(timer_info)
-    end
-    if not no_event then
-        UIManager:broadcastEvent(Event:new("UpdateFooter", true))
+    else
+        return ""
     end
 end
 
-function ReadTimer:removeAdditionalHeaderContent()
-    self.ui.crelistener:removeAdditionalHeaderContent("\u{23F2}") -- ⏲ timer symbol
-    UIManager:unschedule(ReadTimer.addAdditionalHeaderFooterContent)
+function ReadTimer:addAdditionalHeaderContent()
+    self.ui.crelistener:addAdditionalHeaderContent({func=additional_header_content_func, this=self})
```

Comment:
If you define that function in :init() as `self.additional_header_content_func = function() -- you can use self here end`, you don't need to pass `this=self`.
Which will make your add and remove method symetric (instead of one accepting a table and the other a func).